### ISSUE TITLE
Consider new data structure for mapping view ids to objects.

### ISSUE THREAD

**Issue by pykong** [SublimeLinter/SublimeLinter#734]
Right now multiple dictionaries exist mapping `view ids` to other objects.
A comprehensive list of those can be found in `sublime_linter.py`.
```
            self.loaded_views,
            self.linted_views,
            self.view_syntax,
            persist.errors,
            persist.highlights,
            persist.view_linters,
            persist.views,
            persist.last_hit_times
```
Regarding code quality we may consider that data structure bad.
For example, when deleting a view, SL needs to iterate over a list of those dictionaries to individually remove any reference to that deleted view.

Potential fixes are:
1. Merge the current dicts into a single one. Organized by view ids.
2. More preferably extend the view settings, to include the data currently held in individual dict.

**Comment by FichteFoll** [comment]
Note that it may or may not be an advantage that view-specific settings are persisted across sessions.

**Comment by kaste** [comment]
Wait, wait. What are *view specific settings*? 

Does that mean, if we store the last lint result in a special way, we can reuse it (say file mtime did not change or whatever) and draw the regions etc immediately?

**Comment by FichteFoll** [comment]
`view.settings()` is a settings object specific to each view with a fallback to syntax-specific, project-specific user and default preferences. Any changes made to it are persisted across sessions.

**Comment by kaste** [comment]
That sounds really interesting.

**Comment by braver** [comment]
Fixed in #898
